src/Pyz/Client/Training/Storage/PriceItemStorageReader.php

```diff
@@ -0,0 +1,97 @@
+<?php declare(strict_types = 1);
+
+namespace Pyz\Client\Training\Storage;
+
+use Generated\Shared\Transfer\ProductViewTransfer;
+use Generated\Shared\Transfer\SynchronizationDataTransfer;
+use Generated\Shared\Transfer\TrainingStorageItemTransfer;
+use Pyz\Client\Training\Dependency\Client\StorageClientBridgeInterface;
+use Pyz\Client\Training\Dependency\Service\SynchronizationServiceBridgeInterface;
+use Pyz\Shared\Training\TrainingConstants;
+
+class PriceItemStorageReader implements PriceItemStorageReaderInterface
+{
+    /** @var \Pyz\Client\Training\Dependency\Client\StorageClientBridgeInterface */
+    private $storageClient;
+
+    /** @var \Pyz\Client\Training\Dependency\Service\SynchronizationServiceBridgeInterface */
+    private $synchronizationService;
+
+    /**
+     * @param \Pyz\Client\Training\Dependency\Client\StorageClientBridgeInterface $storageClient
+     * @param \Pyz\Client\Training\Dependency\Service\SynchronizationServiceBridgeInterface $synchronizationService
+     */
+    public function __construct(
+        StorageClientBridgeInterface $storageClient,
+        SynchronizationServiceBridgeInterface $synchronizationService
+    ) {
+        $this->storageClient = $storageClient;
+        $this->synchronizationService = $synchronizationService;
+    }
+
+    /**
+     * @param string $customerNumber
+     * @param int $idProductAbstract
+     * @param string $localeName
+     *
+     * @return \Generated\Shared\Transfer\TrainingStorageItemTransfer|null
+     */
+    public function findData(string $customerNumber, int $idProductAbstract, string $localeName): ?TrainingStorageItemTransfer
+    {
+        $reference = $customerNumber . '_' . $idProductAbstract;
```

Comment:
As mentioned below, just pass `$customerNumber` and `$idProductAbstract` (or even better SKU) to your generator and you are clean here 👍